assets/js/edit.js

```diff
@@ -2740,4 +2740,51 @@ function cf_revisions_ui() {
 
 }
 
+/**
+ * Processor Dynamic Options
+ * @TODO Move this wherever it needs to go.
+ */
+
+window.cfProcessorDynamicOptionsCache = {};
+
+function cfProcessorDynamicOptionsLoad(id) {
+    const select = document.getElementById(id);
+    const callback = select.getAttribute('data-callback');
+    if(window.cfProcessorDynamicOptionsCache[callback]) {
+        console.log('Update option from cache');
+        cfProcessorDynamicOptionsUpdate(select, window.cfProcessorDynamicOptionsCache[callback]);
+    } else {
+        console.log('Fetching options');
+        cfProcessorDynamicOptionsFetch(id);
+    }
+}
+
+function cfProcessorDynamicOptionsRefresh(id) {
+    const select = document.getElementById(id);
+    select.options.length = 0;
+    cfProcessorDynamicOptionsFetch(id)
+}
+
+function cfProcessorDynamicOptionsFetch(id) {
+    const select = document.getElementById(id);
+    select.classList.add('dynamic-fetching');
+    const callback = select.getAttribute('data-callback');
+    const promise = new Promise(window[callback]);
+    promise.then(function(options) {
+        window.cfProcessorDynamicOptionsCache[callback] = options;
+        cfProcessorDynamicOptionsUpdate(select, options);
+    });
+}
 
+function cfProcessorDynamicOptionsUpdate(select, options) {
+    const value = select.getAttribute('data-value');
+    select.options.length = 0;
+    options.map(function(option) {
```

Comment:
Please make sure to add tests to cover then `options` is an empty array and for when it is undefined.